picard/script.py

```diff
@@ -189,13 +233,13 @@ def __init__(self):
         self._function_stack = LifoQueue()
 
     def __raise_eof(self):
-        raise ScriptEndOfFile("Unexpected end of script at position %d, line %d" % (self._x, self._y))
+        raise ScriptEndOfFile(StackItem(line=self._y, column=self._x))
 
     def __raise_char(self, ch):
         #line = self._text[self._line:].split("\n", 1)[0]
         #cursor = " " * (self._pos - self._line - 1) + "^"
         #raise ScriptSyntaxError("Unexpected character '%s' at position %d, line %d\n%s\n%s" % (ch, self._x, self._y, line, cursor))
-        raise ScriptSyntaxError("Unexpected character '%s' at position %d, line %d" % (ch, self._x, self._y))
+        raise ScriptSyntaxError(StackItem(line=self._y, column=self._x), "Unexpected character '%s'")
```

Comment:
```python
raise ScriptSyntaxError(StackItem(line=self._y, column=self._x), "Unexpected character '%s'" % ch)
```

Reply:
This didn't trigger a test error, one should be added.

Reply:
Test updated, and it would now catch that error (if it still existed).  I tested it before fixing the error.